### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by nehcuy**
Added the following to DG:
- Update `SettingUp.md`
- Update `AppointmentObjectDiagram` correctness
- Clean grammatical issues
- Add manual testcases
- Add Appendix on team effort
- Merge conflicts twice 💀
- Increase code quality

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -436,121 +441,107 @@ All get commands are implemented in the following steps:
 4. The model is then updated such that the *filtered* list only displays patients whose details match the query
    arguments of that prefix
 
+#### Appointment Date (/appton)
+
+Getting the list of patients with an appointment on the query date involves the following steps:
+1. Prefix `/appton` is matched using an instance of `GetCommandParser`
+2. A new `GetAppointmentByDateCommandParser` instance is created and parses the user input (specifically the date inputted)
+3. A `GetAppointmentByDateCommand` instance containing the date of the appointment is created and returned
+4. The `GetAppointmentByDateCommand` is executed, accessing the list of `PastAppointment` of the specified patient
```
Comment (Charles1026) [inline docs/DeveloperGuide.md:450]: This part sounds abit weird, like only 1 patient's past appointment is checked. 

Maybe can be like "... accesses the list of 'PastAppointment's for every patient, and filters for those who have a past appointment on the specified date."

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -436,121 +441,107 @@ All get commands are implemented in the following steps:
...
-Getting the list of patients in th query appointment date involves the following steps:
-1. prefix "/appton" is matched in `GetCommandParser` class
-2. a new `GetAppointmentByDateCommandParser` instance is created and parsers the user input
-3. a `GetAppointmentByDateCommand` instance is returned
-4. the model is updated such that the *filtered* list only displays patients who have the same query appointment date
+Getting the information of the next of kin of the list of query patient names involves the following steps:
+1. Prefix `/nok` is matched in `GetCommandParser` class
+2. A new `GetNextOfKinCommandParser` instance is created and parses the user input
+3. A `GetNextOfKinCommand` instance is returned
+4. The model is updated such that the *filtered* list only displays queried patients' next of kin details
```
Comment (Charles1026) [inline docs/DeveloperGuide.md:518]: I think there's sth strange abt the tense, everything on top uses query(which Im not sure if it is the correct tense). Best to standardise the tense bah

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -791,14 +758,14 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
   Use case resumes at step 2.
 
-**Use case: Delete a patient's appointment**
+<h3>Use case 5: Removing a patient's next appointment date</h3>
```
Comment (Charles1026) [inline docs/DeveloperGuide.md:761]: This use case was for deleting past appointments but I think the new one works fine also, just an FYI

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1149,14 +1152,154 @@ _
...
+  renaming of classes, methods and variables to fit the context of our application.
+   - Implementing additional details to the `Person` class means refactoring of multiple other classes which
+       involve the `Person` class.
+   - Refactoring of test cases were also required to include the new details of the `Person` class.
+
+### Achievements
+
+Here are some of the features we achieved:
+
+1. Past and future appointments for patients
```
Comment (Charles1026) [inline docs/DeveloperGuide.md:1301]: The tense here is abit weird, shld be like
Creating Past and Future appointments...
Setting Inpatient...

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1149,14 +1152,154 @@ _
...
+2. Inpatient and outpatient patient types
+3. Filtering of patients according to given prefix
+4. Next of kin of patients
+5. Long-term medication of patients
```
Comment (Charles1026) [inline docs/DeveloperGuide.md:1305]: Also if it is missing an EOF whitespace line better to add 1 just in case

**Review by Charles1026 — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (nehcuy) [reply docs/DeveloperGuide.md:518]: Got it, will standardise tense used across the `get` commands.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (nehcuy) [reply docs/DeveloperGuide.md:761]: Changed deliberately to make Use Cases sound more relatable to an actual event. Thanks for heads up!

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (nehcuy) [reply docs/DeveloperGuide.md:1305]: Yeap thanks for the reminder! Confirmed there is an EOF whitespace.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (nehcuy) [reply docs/DeveloperGuide.md:450]: Fixed. Good catch!

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (nehcuy) [reply docs/DeveloperGuide.md:1301]: Resolved.

**Review by Charles1026 — APPROVED** [review]
Great job formatting all of the DG, LGTM

**Pull request merged**